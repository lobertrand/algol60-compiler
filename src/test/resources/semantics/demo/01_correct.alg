begin
    comment Procedure declaration;
    real procedure max(a, b);
    value a, b; real a, b;
    begin
        comment If-expression;
        max := if a > b then a else b
    end;
    real result;

    comment Procedure call;
    result := max(12.5, 3)
end
