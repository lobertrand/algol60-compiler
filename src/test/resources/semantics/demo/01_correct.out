Algol60> Lexical analysis successful
Algol60> Syntactic analysis successful
┌───────────────────────────────────────────────┐
│ level: 0, tableNumber: 0                      │
├───────────────────────────────────────────────┤
│ Procedure: void outstring(integer, string)    │
│ Procedure: void outboolean(integer, boolean)  │
│ Procedure: void outinteger(integer, integer)  │
│ Procedure: void outreal(integer, real)        │
│ Procedure: void inreal(integer, real)         │
│ Procedure: void ininteger(integer, integer)   │
│ Procedure: integer entier(real)               │
└───────────────────────────────────────────────┘
    ┌──────────────────────────┐
    │ level: 1, tableNumber: 1 │
    ├──────────────────────────┤
    │ Variable: integer a      │
    │ Variable: integer b      │
    │ Label: switchEnd         │
    └──────────────────────────┘
        ┌────────────────────────────────┐
        │ level: 2, tableNumber: 2       │
        ├────────────────────────────────┤
        │ Switch: state [first, second]  │
        │ Variable: boolean chooseFirst  │
        │ Label: first                   │
        │ Label: second                  │
        └────────────────────────────────┘

Algol60> Semantic analysis successful
